rust/tensors.rs

```diff
@@ -118,6 +126,37 @@ impl<'a> TensorDescriptor<'a> {
     }
 }
 
+impl fmt::Display for ElementType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let element_kind = String::from(match self {
+            ElementType::Bool => "b",
+            ElementType::UInt8 => "u8",
+            ElementType::Int8 => "i8",
+            ElementType::Int16 => "i16",
+            ElementType::Int32 => "i32",
+            ElementType::Int64 => "i64",
+            ElementType::Float16 => "f16",
+            ElementType::Float32 => "f32",
+            ElementType::Float64 => "f64",
+            ElementType::Complex64 => "c64",
+            ElementType::Complex128 => "c128",
+            ElementType::String => "string",
+            _ => "?"
+        });
+
+        write!(f, "{}", element_kind)
+    }
+}
+
+impl fmt::Display for TensorDescriptor<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}: {}[{}]",
+                self.name.to_str().unwrap(),
+                self.element_type,
+                self.shape.iter().map(|&i| i.to_string()).join(","))
```

Comment:
The [`Itertools::join()`](https://docs.rs/itertools/0.10.1/itertools/trait.Itertools.html#method.join) method can work on any item type as long as it implements `std::fmt::Display`, so we should be able to skip some unnecessary `to_string()` calls here.

```suggestion
        write!(f, "{}: {}[{}]",
                self.name.to_str().unwrap(),
                self.element_type,
                self.shape.iter().join(","))
```